enterprise_access/apps/customer_billing/constants.py

```diff
@@ -72,3 +72,20 @@ class CheckoutIntentState(StrEnum):
     ERRORED_STRIPE_CHECKOUT = 'errored_stripe_checkout'
     ERRORED_PROVISIONING = 'errored_provisioning'
     EXPIRED = 'expired'
+
+
+ALLOWED_CHECKOUT_INTENT_STATE_TRANSITIONS = {
+    CheckoutIntentState.CREATED: [
+        CheckoutIntentState.PAID,
+        CheckoutIntentState.ERRORED_STRIPE_CHECKOUT,
+        CheckoutIntentState.EXPIRED,
+    ],
+    CheckoutIntentState.PAID: [
+        CheckoutIntentState.FULFILLED,
+        CheckoutIntentState.ERRORED_PROVISIONING,
+    ],
+    CheckoutIntentState.ERRORED_STRIPE_CHECKOUT: [CheckoutIntentState.PAID],
+    CheckoutIntentState.ERRORED_PROVISIONING: [CheckoutIntentState.PAID],
+    CheckoutIntentState.FULFILLED: [],
+    CheckoutIntentState.EXPIRED: [],
```

Comment:
I drew EXPIRED as a terminal state in the state diagram, but but I didn't think too hard about it until now.

Q: After expiry are we always just treating user retries as new checkout intents?  Or should we attempt to re-use existing expired checkout intents?

I'm leaning on just creating new ones and treating expired CheckoutIntent records as immutable (no change to your PR).

Reply:
Oh, good point! Right now the model actually has a 1-1 relationship to `User`. I kind of like that pattern (which means we'd reuse existing expired intents for the same user), so I'll change this to allow `EXPIRED -> CREATED` and update the tests.